barrels/LogMonkey/LogTests.mc

```diff
@@ -0,0 +1,246 @@
+using Toybox.Test as Test;
+using Toybox.Lang as Lang;
+using LogMonkey as Log;
+
+module LogMonkey {
+
+    // Wrap the unit tests in a module annotated with :test so it will get compiled
+    // out for non unit test builds.
+    (:test)
+    module Test {
+
+        //! This class mocks the Toybox.System.println() interface but instead
+        //! of printing a value saves it to a string value which can be retrieved
+        //! and examed to verify log results are correct.
+        class MockSys {
+
+            private var mStreamContents;
+
+            //! A mock of Toybox.System.println() function
+            //! @param [Toybox::Lang::Object] message The message to print/log
+            public function println(message) {
+                if (mStreamContents == null) {
+                    mStreamContents = message + "\n";
+                } else {
+                    mStreamContents = mStreamContents + message + "\n";
+                }
+            }
+
+            //! Gets the contents of the fake print stream.
+            //! @return [Toybox::Lang::String] The contents that have been "printed" so far
+            public function getStreamContents() {
+                return mStreamContents;
+            }
+
+        }
+
+        // This class is used within the unit tests module to create an Exception
+        // object with a custom message.
+        class TestException extends Lang.Exception {
+            function initialize() {
+                Exception.initialize();
+                mMessage = "This is a test Exception";
```

Comment:
It appears as though you're accessing an undocumented member here. Should we instead be overriding `getErrorMessage()` to return this literal string?

Reply:
Done